### PR TITLE
Store term positions in index

### PR DESCRIPTION
**Pull request by SebastianSchmidl**
need for assignment2 (search phrases)

### CODE REVIEW

File: cse/lang/__init__.py [hunk 1]
```diff
@@ -47,9 +47,9 @@ def useCustomStopwordList(self, stopwords):
         return self
 
 
-    def appendToStopwordList(self, stopwords):
+    def appendToStopwordList(self, *stopwords):
         self.__useStopwords = True
-        self.__stopwords.append(stopwords)
+        self.__stopwords = self.__stopwords + list(stopwords)
```
Comment (Benedikt1992) [inline cse/lang/__init__.py:52]: It's not directly related with the token position, is it?

File: cse/lang/NltkStemmer.py [hunk 2]
```diff
@@ -19,9 +19,9 @@ def porter2():
...
-        return [self.__stemmerType.stem(token) for token in tokens]
+    def processAll(self, tokenTuples):
+        return [(self.__stemmerType.stem(token), position) for token, position in tokenTuples]
 
 
-    def process(self, token):
-        return self.__stemmerType.stem(token)
\ No newline at end of file
+    def process(self, tokenTuple):
+        return (self.__stemmerType.stem(tokenTuple[0]), tokenTuple[1])
```
Comment (Benedikt1992) [inline cse/lang/NltkStemmer.py:27]: Isn't `tokenTuple[1]` the positon o the token in `tokenTuple[0]`

File: cse/lang/NltkLemmatizer.py [hunk 3]
```diff
@@ -14,10 +14,10 @@ def nltkDefault():
...
-    def processAll(self, tokens):
-        return [self.__lemmatizerType.lemmatize(token) for token in tokens]
+    def processAll(self, tokenTuples):
+        return [(self.__lemmatizerType.lemmatize(token), position) for token, position in tokenTuples]
 
 
-    def process(self, token):
-        return self.__lemmatizerType.lemmatize(token)
+    def process(self, tokenTuple):
+        return (self.__lemmatizerType.lemmatize(tokenTuple[0]), tokenTuple[1])
```
Comment (Benedikt1992) [inline cse/lang/NltkLemmatizer.py:22]: Same as stemmer

File: cse/indexing/DeltaPostingListIndex.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ class DeltaPostingListIndex(object):
     __numCommentIds = 0
 
 
-    def __init__(self, cidSize=35): # import sys; cid = "14d2c537-d2ed-4e36-bf3d-a26f62c02370"; assert(sys.getsizeof(cid) == 85)
+    def __init__(self, cidSize=42): # import sys; cid = "14d2c537-d2ed-4e36-bf3d-a26f62c02370"; assert(sys.getsizeof(cid) == 85)
```
Comment (Benedikt1992) [inline cse/indexing/DeltaPostingListIndex.py:17]: I count 36 characters

Comment (SebastianSchmidl) [reply cse/indexing/DeltaPostingListIndex.py:17]: bytes used to store the string in memory, not chars
but I adopted the size to match more closely to the needed size on disk (via experiments)

File: cse/lang/__init__.py (hunk 1 above)
Comment (SebastianSchmidl) [reply cse/lang/__init__.py:52]: yeah. That was a bug in the previous code. (related to the mentioned caveates of dynamically typed languages)
`append` only accepts one element for the list
`+` allows concatenation of two lists

the method's definition does not specify explicitly what parameter type is needed, but the name of the method let someone assume a list is needed, therefore I changed it to handle both ways

File: cse/lang/NltkStemmer.py (hunk 2 above)
Comment (SebastianSchmidl) [reply cse/lang/NltkStemmer.py:27]: I don't understand your question. A `tokenTuple` in the preprocessing pipeline consists of two elements: `(token, position)`, `tokenTuple[0]` refers to `token` and `tokenTuple[1]` refers to `position`

To be consistent over all preprocessing steps I decided to let every step expect a token tuple as a parameter. Therefore the return type of the processing result must also be a token tuple. The most steps don't touch the position of the token, therefore it is only passed to the result tuple via `return (<sth>, tokenTuple[1])`

Comment (Benedikt1992) [reply cse/lang/NltkStemmer.py:27]: Ok my fault. I read `.stem(tokenTuple[0], tokenTuple[1])`

**Review by Benedikt1992 — APPROVED** [review]

**Pull request merged**